lessons/git/intro/slides-2017.Rmd

```diff
@@ -0,0 +1,305 @@
+```bash
+git remote add remote_name remote_url
+```
+
+Before you do this, you need to have an account on GitHub and a repo on it
+
+# Configuring your repo for use with GitHub on the cloud
+* Copy the repo url from the repo creation page and use it in
+```bash
+git remote add origin copy_url_here
+```
+
+* Check your remote configuration again
+```bash
+git remote -v
+```
+
+# Adding new files to a repo
+* create a text file with
+    + your name
+    + program
+    + research area
+* save the file as `my_cred` in the git repo you just created
+
+# Visualize changes to files
+You can see the changes you made to this file in GitHub
+\begin{block}
+{New files} on GitHub can be seen with a \textcolor{green}{green '+'} to their right 
+\end{block}
+
+This indicates the file is new and is `untracked` by git
+
+\begin{block}
+{Old modified files} can be seen with a \textcolor{gray}{gray circle} to their right
+\end{block}
+
+This indicates the file is already `tracked` and there are `uncommited` changes
+```bash
+git status
+``` 
+can give information on `tracked`, `untracked` and `indexed` files
+
+# Adding Files to git index
+* add the file to the git index
+```bash
+git add ~/Documents/new_git_repo/my_cred
+# you can also use wildcards instead of full name
+git add *my_cred
+```
+* check file status again 
+```bash
+git status
+```
+Notice the file has changed from \textcolor{red}{red} to \textcolor{green}{green}
+
+* let us `commit` this change to repo history 
+```bash
+git commit -m "initial commit with my info"
+```
+* Now your file is `tracked` by git
+
+# `push` to your new repo on GitHub
+You can move all your **committed** changes to the remote repo
+```bash
+git push origin
+```
+You can also get all **commits** from the remote repo
+```bash
+git pull origin
+```
+Make sure your `remote` is set before you do this
+
+# Cloning an Existing `git` repo
+We can copy or clone existing repositories from GitHub using
+```bash
+git clone
+```
+Let us do this for the UofT Coders repo available at
+[UofT Coders git repo](https://github.com/UofTCoders/studyGroup)
+
+Select the green `Clone or Download` button and select ` copy to clipboard`
+
+In your shell/bash move to a directory of your choice and clone the repo
+```bash
+cd ~/Documents
+# if you are using git posh (in powershell) 
+# use the right click to paste the copied url
+git clone https://github.com/UofTCoders/studyGroup
+```
+
+# Modify a existing file and commit changes
+Go to the directory containing the example file
+```bash
+cd C:/Users/shyam/Documents/studyGroup/lessons/git/intro
+# see all files in the directory
+ls
+```
+Open file `example_file.md` in any text editor
+
+To open files in bash/shell
+```bash
+editor_name file_name
+gitpad example_file.md
+```
+add a new line to this file starting with an asterisk(*)
+
+save file and close
+
+# Visualize changes to file
+You can visualize on GitHub as before or 
+
+You can also use
+```bash
+git diff example_file.md
```

Comment:
This is not really equivalent to looking at GitHub, since that can only show you diffs between commits, whereas this command by default is showing you uncommitted & unstaged changes.

Reply:
Oh, are you talking about *GitHub Desktop* here? I don't know if you should be using that if you only asked Windows to install it.

Reply:
True. What would you suggest regrading GitHub desktop though?